### PR TITLE
Deprecate HttpApp

### PR DESCRIPTION
**Pull request by ennru**
References #3155

### CODE REVIEW

File: docs/src/main/paradox/routing-dsl/index.md [hunk 1]
```diff
@@ -57,8 +56,6 @@ Scala
 Java
```
Comment (jrudolph) [inline docs/src/main/paradox/routing-dsl/index.md:56]: Did you keep the lines above intentionally?

File: docs/src/test/scala/docs/http/scaladsl/HttpAppExampleSpec.scala [hunk 2]
```diff
@@ -4,10 +4,12 @@
 
 package docs.http.scaladsl
 
+import com.github.ghik.silencer.silent
 import docs.CompileOnlySpec
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
 
+@silent("deprecated")
```
Comment (jrudolph) [inline docs/src/test/scala/docs/http/scaladsl/HttpAppExampleSpec.scala:12]: Is there a particular reason to keep the docs examples around?

Comment (ennru) [reply docs/src/test/scala/docs/http/scaladsl/HttpAppExampleSpec.scala]: Dropped them now.

File: docs/src/main/paradox/routing-dsl/index.md (hunk 1 above)
Comment (ennru) [reply docs/src/main/paradox/routing-dsl/index.md]: How could I not see this? Removed.

**Pull request merged**